crypto/chacha/chacha_riscv.c

```diff
@@ -44,12 +44,21 @@ void ChaCha20_ctr32_zvkb(unsigned char *out, const unsigned char *inp,
                          size_t len, const unsigned int key[8],
                          const unsigned int counter[4]);
 
+void ChaCha20_ctr32_v(unsigned char *out, const unsigned char *inp,
+                      size_t len, const unsigned int key[8],
+                      const unsigned int counter[4]);
+
 void ChaCha20_ctr32(unsigned char *out, const unsigned char *inp, size_t len,
                     const unsigned int key[8], const unsigned int counter[4])
 {
-    if (RISCV_HAS_ZVKB() && riscv_vlen() >= 128) {
-        ChaCha20_ctr32_zvkb(out, inp, len, key, counter);
-    } else {
+    if (riscv_vlen() >= 128) {
+        if (RISCV_HAS_ZVKB()) {
+            ChaCha20_ctr32_zvkb(out, inp, len, key, counter);
+        } else {
```

Comment:
Will this new code work on all RISCV64 CPUs where riscv_vlen() returns >=128? I.e., there is no need to add any further condition?

Reply:
I think Yes.